guides/introduction/templates.md

```diff
@@ -0,0 +1,202 @@
+
+The `link` function creates HTML hyperlinks. It takes care of specifying
+the `rel` and `target` when necessary, and it supports both full URLs or
+paths relative to the input folder. You should always use the link
+function to create hyperlinks between files, otherwise deploys to
+subfolders will not work. For instance:
+
+```slim
+= link "Home", to: "/"
+= link "Blog", to: "https://example.org"
+```
+
+### Including images
+
+Image support is still (get it?) rudimentary: it can generate images of
+different sizes, and apply transformations to them. By default, it relies
+on [ImageMagick][imagemagick], but if you add [`imageflow_ex`][imageflow]
+as a dependency it will use that. The reason we do not use `imageflow_ex`
+by default is that it requires Rust. If you want to use ImageMagick, you
+also need to have it installed in your system.
+
+`responsive_image` receives an image and some opts and generates an HTML
+`img` which with the correct `src` and `sizes` for the different image
+sizes it generates.
+
+To insert an image use the `responsive_image` function, which accepts
+a list of keyword arguments. To generate an image, Still only cares about the
+`:image_opts` key, everything else will be rendered as an attribute on the
+`img` tag. `:image_opts` has to be a map where you can specify `:sizes`
+and `:transformations`.
+
+`:sizes` is optional and by default we take the image's width to generate
+3 smaller images. If you specify an array of integers, it will use that
+instead.
+
+`:transformations` is also optional, and it's passed down to either
+[ImageMagick](imagemagic-cli-option] or [Imageflow][imageflow-docs], see
+each project's documentation for more information.
+
+A responsive image can be generated like this:
+
+```eex
+<%= responsive_image("example.jpg", image_opts: %{transformations: [grayscale: "Rec709Luma"]}) %>
+```
+
+```slime
+= responsive_image("cover.jpg", image_opts: %{transformations: [color_filter: "grayscale_bt709"]}, class: "cover")
+```
+
+[imagemagick]: https://imagemagick.org/
+[imageflow]: https://github.com/imazen/imageflow
+[imagemagic-cli-option]: https://imagemagick.org/script/command-line-options.php
+[imageflow-docs]: https://docs.imageflow.io/
+
+### Custom helpers
+
+You can call any module from the templates, but if you defined some modules in the config, their functions are imported to the templates:
+
+```elixir
+config :still,
+  view_helpers: [Your.Module]
+```
+
+## Layouts
+
+Layouts are templates that wrap other content. To use a layout, set the
+`layout` key in front matter. For instance:
+
+```slime
+---
+layout: _layout.slime
+---
+
+h1 About Page
+```
+
+This will look for a `_layout.slime` in your input folder.
+
+The layout file must print the children variable. For instance:
+
+```slime
+doctype html
+html
+  body
+    = @children
+```
+
+Notice that `_layout.slime` starts with an underscore. It's necessary
+because we don't want to compile the layout file by itself. Any file
```

Comment:
```suggestion
Notice that `_layout.slime` starts with an underscore. **This is necessary
since we don't want to compile the layout file by itself**. Any file
```